src/main/scala/org/broadinstitute/dsde/agora/server/business/PermissionBusiness.scala

```diff
@@ -1,72 +1,114 @@
+  def insertEntityPermission(entity: AgoraEntity, requester: String, accessObject: AccessControl): Int = {
+    // Inserts can result in an edit in the DAO layer
+    val entityACLs = getEntityACLs(entity, requester)
+    checkSameRequesterAndPermissions(entityACLs.find(acl => acl.user.equals(requester)), accessObject)
+    authorizeEntityRequester(entityACLs, entity, requester)
     AgoraEntityPermissionsClient.insertEntityPermission(entity, accessObject)
   }
 
-  def batchEntityPermission(entity: AgoraEntity, requester: String, accessObjectList: List[AccessControl]) = {
+  def batchEntityPermission(entity: AgoraEntity, requester: String, accessObjectList: List[AccessControl]): Int = {
     accessObjectList.map(accessObject => insertEntityPermission(entity, requester, accessObject)).sum
   }
 
-  def editEntityPermission(entity: AgoraEntity, requester: String, accessObject: AccessControl) = {
-    authorizeEntityRequester(entity, requester)
+  def editEntityPermission(entity: AgoraEntity, requester: String, accessObject: AccessControl): Int = {
+    val entityACLs = getEntityACLs(entity, requester)
+    checkSameRequesterAndPermissions(entityACLs.find(acl => acl.user.equals(requester)), accessObject)
+    authorizeEntityRequester(entityACLs, entity, requester)
     AgoraEntityPermissionsClient.editEntityPermission(entity, accessObject)
 
   }
 
-  def deleteEntityPermission(entity: AgoraEntity, requester: String, userToRemove: String) = {
-    authorizeEntityRequester(entity, requester)
+  def deleteEntityPermission(entity: AgoraEntity, requester: String, userToRemove: String): Int = {
+    checkSameRequester(requester, userToRemove)
+    authorizeEntityRequester(getEntityACLs(entity, requester), entity, requester)
     AgoraEntityPermissionsClient.deleteEntityPermission(entity, userToRemove)
   }
 
-  def authorizeNamespaceRequester(entity: AgoraEntity, requester: String) = {
-    if (!NamespacePermissionsClient.getNamespacePermission(entity, requester).canManage &&
-        !NamespacePermissionsClient.getNamespacePermission(entity, "public").canManage)
-      throw new NamespaceAuthorizationException(AgoraPermissions(Manage), entity, requester)
+
+  private def authorizeNamespaceRequester(acls: Seq[AccessControl], entity: AgoraEntity, requester: String): Unit = {
+    if (!acls.exists(_.roles.canManage))
+      throw NamespaceAuthorizationException(AgoraPermissions(Manage), entity, requester)
+  }
+
+  private def authorizeEntityRequester(acls: Seq[AccessControl], entity: AgoraEntity, requester: String): Unit = {
+    if (!acls.exists(_.roles.canManage))
+      throw AgoraEntityAuthorizationException(AgoraPermissions(Manage), entity, requester)
+  }
+
+  private def checkSameRequester(requester: String, userToModify: String):Unit = {
+    if (requester.equalsIgnoreCase(userToModify)) {
+      val m = "Unable to remove access control for current user"
+      throw AgoraException(m, new Exception(m), Conflict)
+    }
+  }
```

Comment:
I know this is currently only called by deletes - I'd still prefer it had the generic message "Unable to modify..." to make it easier to call from elsewhere in the future.

Also, if we *really* want to make this work like rawls, rawls returns a 400 with "You may not change your own permissions". Might be worthwhile switching to that - eventually we could use the same UI components to handle them.